primitives/src/asset_identifier.rs

```diff
@@ -256,4 +277,32 @@ mod tests {
         );
         assert_eq!(AssetIdentifier::lei(*b"549300GFXDSN7JDUSN34"), None);
     }
+
+    #[test]
+    fn figi() {
+        assert_eq!(
+            AssetIdentifier::figi(*b"BBG000BLNQ16"),
+            Some(AssetIdentifier::FIGI(*b"BBG000BLNQ16"))
+        );
+        assert_eq!(
+            AssetIdentifier::figi(*b"NRG92C84SB39"),
+            Some(AssetIdentifier::FIGI(*b"NRG92C84SB39"))
+        );
+        assert_eq!(
+            AssetIdentifier::figi(*b"BBG0013YWBF3"),
+            Some(AssetIdentifier::FIGI(*b"BBG0013YWBF3"))
+        );
+        assert_eq!(
+            AssetIdentifier::figi(*b"BBG00H9NR574"),
+            Some(AssetIdentifier::FIGI(*b"BBG00H9NR574"))
+        );
+        assert_eq!(
+            AssetIdentifier::figi(*b"BBG00094DJF9"),
+            Some(AssetIdentifier::FIGI(*b"BBG00094DJF9"))
+        );
+        assert_eq!(
+            AssetIdentifier::figi(*b"BBG00024DJF9"),
+            None
+        );
```

Comment:
Some more invalid inputs would be nice and maybe a comment on why it is invalid.